### PR TITLE
Deprecate Either.toValidated() and Either.toValidatedNel() functions

### PR DESCRIPTION
**Pull request by Zordid**
Due to the deprecation of `Validated` and `ValidatedNel` it seems reasonable to symmetrically also deprecate the functions `Either.toValidated()` and `Either.toValidatedNel()`.
This way, users see that they should avoid creating `Validated` objects.

Symmetrically, `Validated.fromEither(...)` had already been deprecated before.

### CODE REVIEW

**Comment by Zordid** [comment]
@nomisRev in case you did not find it ;-)

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Validated.kt [hunk 1]
```diff
@@ -1358,4 +1358,4 @@ public inline fun <E> E.invalidNel(): ValidatedNel<E, Nothing> =
 internal const val ValidatedDeprMsg = "Validated functionally is being merged into Either.\n"
 
 private const val DeprAndNicheMsg =
-  "Validated functionaliy is being merged into Either, but this API is niche and will be removed in the future. If this method is crucial for you, please let us know on the Arrow Github. Thanks!\n https://github.com/arrow-kt/arrow/issues\n"
+  "Validated functionally is being merged into Either, but this API is niche and will be removed in the future. If this method is crucial for you, please let us know on the Arrow Github. Thanks!\n https://github.com/arrow-kt/arrow/issues\n"
```
Comment (nomisRev) [inline arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Validated.kt:1361]:
```suggestion
  "Validated functionality is being merged into Either, but this API is niche and will be removed in the future. If this method is crucial for you, please let us know on the Arrow Github. Thanks!\n https://github.com/arrow-kt/arrow/issues\n"
```

**Review by nomisRev — APPROVED** [review]
LGTM, thanks @Zordid for testing the migration and contributing to Arrow 🙌

**Review by i-walker — APPROVED** [review]

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Either.kt [hunk 2]
```diff
@@ -2297,7 +2296,7 @@ public operator fun <A : Comparable<A>, B : Comparable<B>> Either<A, B>.compareT
   ReplaceWith("Either.zipOrAccumulate({ a, bb -> SGA.run { a.combine(bb) }  }, this, b) { a, bb -> SGB.run { a.combine(bb) } }")
 )
 public fun <A, B> Either<A, B>.combine(SGA: Semigroup<A>, SGB: Semigroup<B>, b: Either<A, B>): Either<A, B> =
-  Either.zipOrAccumulate({ a, bb -> SGA.run { a.combine(bb) }  }, this, b) { a, bb -> SGB.run { a.combine(bb) } }
+  Either.zipOrAccumulate({ a, bb -> SGA.run { a.combine(bb) } }, this, b) { a, bb -> SGB.run { a.combine(bb) } }
```
Comment (serras) [inline arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Either.kt:2299]:
```suggestion
Either.zipOrAccumulate(SGA::combine, this, b, SGB::combine)
```

**Review by serras — APPROVED** [review]
Thanks very much!

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Validated.kt (hunk 1 above)
Comment (Zordid) [reply arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Validated.kt:1361]: Haha, I did not know what is was meant to be - I only knew it was somehow wrong... ;-)

**Comment by Zordid** [comment]
If somebody would like to merge, that would be great - so far I cannot merge myself. Thanks!

**Pull request merged**